### DISCUSSION TITLE
How to obtain the predicted values of validation set?

### DISCUSSION THREAD

**Discussion by phys-chem** [mir-group/nequip#376] [Q&A]
I run nequip-evaluate --train-dir /my_path. How to get the predicted values? Thank you very much!
As the title, I get these outputs:

Using device: cpu
Loading model... 
loaded model from training session
Loading original dataset...
Loaded validation_dataset specified in config.yaml.
Using origial validation dataset (332 frames) minus validation set frames (332 frames), yielding a test set size of 0 frames.
Starting...
0it [00:00, ?it/s]

--- Final result: ---

**Comment by Eipgen** [comment]
it seems that yours `n_train` and `n_val` are same. if you want to check you training set error, you can follow that.

1. copy a config yaml file and set `n_val: 0`
2. then add `--dataset-config your_copied_config_yaml` when use `nequip-evaluate`
3. also you can use `--output` and `--log` for detail results

**Comment by phys-chem** [comment]
Thank you very much! --output tag gives me what I want. If there is an example for the ASE interface, it will be more convenient.

**Comment by Linux-cpp-lisp** [comment]
@Eipgen is right that `--output` is the correct key (you can also use `--output-fields` if you want to output non-standard fields for analysis, such as the final features, for example).  Thanks @Eipgen for responding!

For the ASE interface, see https://github.com/mir-group/nequip/discussions/115, but please note that the ASE interface is not optimal from a performance point of view if you want to run a specific preexisting group of frames, since it cannot batch calculations like `nequip-evaluate` does.
